vulkano/src/memory/device_memory.rs

```diff
@@ -1388,6 +1419,18 @@ pub struct MemoryMapInfo {
     /// [`non_coherent_atom_size`]: crate::device::DeviceProperties::non_coherent_atom_size
     pub size: DeviceSize,
 
+    /// The address in host memory to map to.
+    ///
+    /// Requires [`DeviceExtensions::ext_map_memory_placed`] and
+    /// [`DeviceFeatures::memory_map_placed`] to be enabled.
+    ///
+    /// Must align with [`DeviceProperties::min_placed_memory_map_alignment`].
+    ///
+    /// [`DeviceExtensions::ext_map_memory_placed`]: crate::device::DeviceExtensions::ext_map_memory_placed
+    /// [`DeviceFeatures::memory_map_placed`]: crate::device::DeviceFeatures::memory_map_placed
+    /// [`DeviceProperties::min_placed_memory_map_alignment`]: crate::device::DeviceProperties::min_placed_memory_map_alignment
+    pub placed_address: Option<*mut c_void>,
```

Comment:
```suggestion
    pub placed_address: Option<NonNull<c_void>>,
```

The pointer can't be null, and this allows using the niche for the `Option`.

Reply:
Good idea